scripts/apps/search/components/ItemListAngularWrapper.tsx

```diff
@@ -50,6 +65,20 @@ export class ItemListAngularWrapper extends React.Component<IProps, IState> {
         this.multiSelect = this.multiSelect.bind(this);
 
         this.abortController = new AbortController();
+        this.eventListenersToRemoveBeforeUnmounting = [];
+
+        this.handleContentChanges = throttleAndCombineArray(
+            (changes) => {
+                getAndMergeRelatedEntitiesUpdated(
+                    this.state.relatedEntities,
+                    changes,
+                    this.abortController.signal,
+                ).then((relatedEntities) => {
+                    this.setState({relatedEntities});
```

Comment:
would it actually return the same instance if there were no changes to related entities? and thus setState wouldn't force a re-render on each event

Reply:
A new instance is returned every time. If there are no changes to an entity - websocket events don't fire and this code doesn't run.

Reply:
Right, it fires when any entities change, not only related ones. As you pointed out there is filtering and in case related entities didn't change, the same instance is returned.